src/Toolkit.Forms/OverviewMap/OverviewMap.cs

```diff
@@ -0,0 +1,219 @@
+
+using Esri.ArcGISRuntime.Mapping;
+using Esri.ArcGISRuntime.Symbology;
+using Esri.ArcGISRuntime.Toolkit.UI.Controls;
+using Esri.ArcGISRuntime.Xamarin.Forms;
+using Xamarin.Forms;
+using Xamarin.Forms.Xaml;
+
+namespace Esri.ArcGISRuntime.Toolkit.Xamarin.Forms
+{
+    /// <summary>
+    /// Defines a small "overview" (or "inset") map displaying a representation of the attached <see cref="GeoView"/>'s current viewpoint.
+    /// </summary>
+    public class OverviewMap : TemplatedView
+    {
+        private OverviewMapController? _controller;
+        private static readonly ControlTemplate DefaultControlTemplate;
+
+        private MapView? _overviewMapView;
+
+        static OverviewMap()
+        {
+            string template = @"<ControlTemplate xmlns=""http://xamarin.com/schemas/2014/forms""
+                                                 xmlns:x=""http://schemas.microsoft.com/winfx/2009/xaml""
+                                                 xmlns:esri=""clr-namespace:Esri.ArcGISRuntime.Xamarin.Forms;assembly=Esri.ArcGISRuntime.Xamarin.Forms""
+                                                 xmlns:internal=""clr-namespace:Esri.ArcGISRuntime.Toolkit.Xamarin.Forms.Internal;assembly=Esri.ArcGISRuntime.Toolkit.Xamarin.Forms"">
+                                    <Grid>
+                                        <Grid.Resources>
+                                            <internal:LoadStatusToVisibilityConverter x:Key=""LoadStatusToVisibilityConverter"" />
+                                        </Grid.Resources>
+                                        <ActivityIndicator IsRunning=""{Binding Source={x:Reference PART_MapView}, Path=Map.LoadStatus, Converter={StaticResource LoadStatusToVisibilityConverter}, ConverterParameter='Loading'}"" />
+                                        <Label Text=""Map failed to load. Did you forget an API key?"" IsVisible=""{Binding Source={x:Reference PART_MapView}, Path=Map.LoadStatus, Converter={StaticResource LoadStatusToVisibilityConverter}, ConverterParameter='FailedToLoad'}""  />
+                                        <esri:MapView x:Name=""PART_MapView"" IsVisible=""{Binding Source={x:Reference PART_MapView}, Path=Map.LoadStatus, Converter={StaticResource LoadStatusToVisibilityConverter}, ConverterParameter='Loaded'}"" />
+                                    </Grid>
+                                </ControlTemplate>";
+            DefaultControlTemplate = Extensions.LoadFromXaml(new ControlTemplate(), template);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OverviewMap"/> class.
+        /// </summary>
+        public OverviewMap()
+        {
+            ControlTemplate = DefaultControlTemplate;
+        }
+
+        /// <summary>
+        /// <inheritdoc />
+        /// </summary>
+        protected override void OnApplyTemplate()
+        {
+            base.OnApplyTemplate();
+
```

Comment:
If suggest "cleaning up" the old controller here if it isn't null (set AttachedView to null to unhook event handlers, as well as clear out the weak references - perhaps just add a dispose that does this?)